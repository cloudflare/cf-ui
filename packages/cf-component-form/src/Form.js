import React, { PropTypes } from 'react';
import { createComponent } from 'cf-style-container';

const styles = ({ theme }) => ({
  background: theme.background,
  border: theme.border,
  boxShadow: theme.boxShadow
});

const addLayoutProp = (children, layout) =>
  React.Children.map(children, (child, index) => {
    if (React.isValidElement(child)) {
      return React.cloneElement(child, { layout });
    }
    return child;
  });

const Form = ({ children, layout, onSubmit, className }) => (
  <form className={className} onSubmit={onSubmit}>
    {addLayoutProp(children, layout)}
  </form>
);

Form.propTypes = {
  layout: PropTypes.oneOf(['horizontal', 'vertical']).isRequired,
  className: PropTypes.string,
  onSubmit: PropTypes.func.isRequired,
  children: PropTypes.node
};

Form.displayName = 'Form';

export default createComponent(styles, Form);
