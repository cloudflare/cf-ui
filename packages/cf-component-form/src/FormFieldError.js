import React from 'react';
import PropTypes from 'prop-types';
import { createComponent } from 'cf-style-container';

const styles = ({ theme }) => ({
  position: theme.position,
  marginTop: theme.marginTop,
  padding: theme.padding,
  border: theme.border,
  fontSize: theme.fontSize,
  fontWeight: theme.fontWeight,
  background: theme.background,
  color: theme.color,
  borderRadius: theme.borderRadius,
  boxShadow: theme.boxShadow,
  '-webkit-font-smoothing': 'antialiased',
  '&::before': {
    content: '""',
    display: 'block',
    position: 'absolute',
    bottom: '100%',
    borderStyle: 'solid',
    borderColor: 'transparent',
    left: '4px',
    borderWidth: '6px',
    borderBottomColor: theme.colorRed
  },
  '&::after': {
    content: '""',
    display: 'block',
    position: 'absolute',
    bottom: '100%',
    borderStyle: 'solid',
    borderColor: 'transparent',
    left: '4px',
    borderWidth: '4px',
    borderBottomColor: theme.colorRed
  },
  '& p': {
    marginTop: 0,
    marginBottom: 0
  },

  '& p + p': {
    marginTop: '0.5em'
  }
});

class FormFieldError extends React.Component {
  render() {
    const { field, validations, className } = this.props;
    if (!field.invalid) {
      return null;
    }

    return (
      <div className={className}>
        {Object.keys(validations).map(validation => {
          return (
            <p key={validation}>
              {validations[validation]}
            </p>
          );
        })}
      </div>
    );
  }
}

FormFieldError.propTypes = {
  field: PropTypes.object.isRequired,
  validations: PropTypes.object.isRequired,
  className: PropTypes.string
};

export default createComponent(styles, FormFieldError);
