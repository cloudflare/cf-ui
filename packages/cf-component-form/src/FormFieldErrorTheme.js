export default baseTheme => ({
  position: 'relative',
  marginTop: '-0.75em',
  padding: '0.5em 1em',
  border: `1px solid  ${baseTheme.colorRedDark}`,
  fontSize: '0.8em',
  fontWeight: baseTheme.weightSemiBold,
  background: baseTheme.colorRed,
  color: baseTheme.colorWhite,
  boxShadow: '0 1px 1px rgba(0, 0, 0, 0.1)'
});
