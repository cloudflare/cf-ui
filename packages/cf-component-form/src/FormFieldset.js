import React from 'react';
import PropTypes from 'prop-types';
import { createComponentStyles } from 'cf-style-container';
import { clearFix } from 'polished';

const mainStyles = ({ theme }) => ({
  margin: theme.main.margin,
  padding: theme.main.padding,
  borderTop: theme.main.borderTop,
  borderLeft: theme.main.borderLeft,
  borderRight: theme.main.borderRight,
  borderBottom: theme.main.borderBottom,
  ...clearFix()
});

const legendStyles = ({ theme, layout }) => ({
  padding: theme.legend.padding,
  marginBottom: theme.legend.marginBottom,
  borderBottom: theme.legend.borderBottom,
  tablet: layout === 'horizontal'
    ? {
        width: '30%',
        float: 'left',
        padding: '1.7em 1rem',
        color: theme.colorGrayDark,
        textAlign: 'right'
      }
    : {}
});

const contentStyles = ({ theme, layout }) => ({
  padding: theme.content.padding,
  border: theme.content.border,
  borderTopWidth: theme.content.borderTopWidth,
  background: theme.content.background,
  tablet: layout === 'horizontal'
    ? {
        width: '70%',
        float: 'left',
        borderTopWidth: 0,
        borderLeftWidth: 0
      }
    : {}
});

class FormFieldset extends React.Component {
  render() {
    const { legend, children, styles } = this.props;
    return (
      <fieldset className={styles.mainStyles}>
        <legend className={styles.legendStyles}>
          {legend}
        </legend>
        <div className={styles.contentStyles}>
          {children}
        </div>
      </fieldset>
    );
  }
}

FormFieldset.propTypes = {
  layout: PropTypes.oneOf(['horizontal', 'vertical']),
  styles: PropTypes.object.isRequired,
  legend: PropTypes.string.isRequired,
  children: PropTypes.node
};

FormFieldset.defaultProps = {
  layout: 'vertical'
};

export default createComponentStyles(
  { mainStyles, legendStyles, contentStyles },
  FormFieldset
);
