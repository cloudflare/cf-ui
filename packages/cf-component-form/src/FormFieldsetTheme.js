import { checkBaseTheme } from 'cf-style-const';

export default baseTheme => {
  checkBaseTheme(baseTheme, 'FormFieldsetTheme');
  return {
    main: {
      margin: 0,
      padding: 0,
      borderTop: 'none',
      borderLeft: 'none',
      borderRight: 'none',
      borderBottom: `1px solid ${baseTheme.colorGrayLight}`
    },
    legend: {
      padding: '1em',
      borderBottom: 'none',
      marginBottom: 0
    },
    content: {
      padding: '1em',
      border: `0 solid ${baseTheme.colorGrayLight}`,
      borderTopWidth: '1px',
      background: '#eee'
    }
  };
};
