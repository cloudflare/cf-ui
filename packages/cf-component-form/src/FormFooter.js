import React, { PropTypes } from 'react';
import { createComponent } from 'cf-style-container';

const styles = ({ theme }) => ({
  padding: theme.padding,
  borderTop: theme.borderTop,
  textAlign: theme.textAlign
});

const FormFooter = ({ className, children }) => (
  <div className={className}>
    {children}
  </div>
);

FormFooter.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string
};

export default createComponent(styles, FormFooter);
