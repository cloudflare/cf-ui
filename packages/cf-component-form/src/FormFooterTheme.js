import { checkBaseTheme } from 'cf-style-container';

export default baseTheme => {
  checkBaseTheme(baseTheme, 'FormFooterTheme');
  return {
    padding: '1em',
    borderTop: `1px solid ${baseTheme.colorGrayLight}`,
    textAlign: 'right'
  };
};
