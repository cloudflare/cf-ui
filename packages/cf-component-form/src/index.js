import FormUnstyled from './Form';
import FormTheme from './FormTheme';
import FormHeader from './FormHeader';
import FormFooterUnstyled from './FormFooter';
import FormFooterTheme from './FormFooterTheme';
import FormFieldsetUnstyled from './FormFieldset';
import FormFieldsetTheme from './FormFieldsetTheme';
import FormLabel from './FormLabel';
import FormFieldErrorUnstyled from './FormFieldError';
import FormFieldErrorTheme from './FormFieldErrorTheme';

import { applyTheme } from 'cf-style-container';

const Form = applyTheme(FormUnstyled, FormTheme);
const FormFieldError = applyTheme(FormFieldErrorUnstyled, FormFieldErrorTheme);
const FormFieldset = applyTheme(FormFieldsetUnstyled, FormFieldsetTheme);
const FormFooter = applyTheme(FormFooterUnstyled, FormFooterTheme);

export {
  Form,
  FormUnstyled,
  FormTheme,
  FormHeader,
  FormFooter,
  FormFooterUnstyled,
  FormFooterTheme,
  FormFieldset,
  FormFieldsetUnstyled,
  FormFieldsetTheme,
  FormLabel,
  FormFieldError,
  FormFieldErrorUnstyled,
  FormFieldErrorTheme
};
