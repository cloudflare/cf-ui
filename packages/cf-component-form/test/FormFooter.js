import React from 'react';
import { FormFooter } from '../src/index';
import { felaSnapshot } from 'cf-style-provider';

test('should render', () => {
  const snapshot = felaSnapshot(<FormFooter>FormFooter</FormFooter>);
  expect(snapshot.component).toMatchSnapshot();
  expect(snapshot.styles).toMatchSnapshot();
});
