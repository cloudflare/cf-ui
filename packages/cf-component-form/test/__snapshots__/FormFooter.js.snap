// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="a b c"
>
  FormFooter
</div>
`;

exports[`should render 2`] = `
"
.a {
  padding: 1em
}

.b {
  border-top: 1px solid #dedede
}

.c {
  text-align: right
}
"
`;
